### PR TITLE
Fix `parse_race_history_chart.py` and `parse_race_lap_chart.py`, add testing

### PR DESCRIPTION
**Pull request by tylerjthomas9**
`Lap` was changed to use `lap_number` instead of just `number` [here](https://github.com/harningle/fia-doc/commit/fa14b51231d98d0c8eabf8ba78f4ab5ea476a99c). I also added quick smoke tests to verify the code works as expected.

### CODE REVIEW

File: parse_race_lap_chart.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def to_json(df: pd.DataFrame):
     session_type = 'R'
 
     # Convert to json
-    df['lap'] = df.apply(lambda x: Lap(number=x['lap'], position=x['position']), axis=1)
+    df['lap'] = df.apply(lambda x: Lap(lap_number=x['lap'], position=x['position'], time=0.0), axis=1)
```
Comment (harningle) [inline parse_race_lap_chart.py:93]: I will probably change the entire race lap data stuff later, as the current parsing really mess a lot of things...

File: parse_pit_stop_summary.py [hunk 2]
```diff
@@ -75,8 +75,12 @@ def to_json(df: pd.DataFrame):
     ).tolist()
     with open('pit_stops.pkl', 'wb') as f:
         pickle.dump(pit_stop_data, f)
-    pass
+    return pit_stop_data
 
 
 if __name__ == '__main__':
-    pass
+    df = parse_pit_stop_summary('race_pit_stop_summary.pdf')
```
Comment (harningle) [inline parse_pit_stop_summary.py:82]: put them into separate tests using things like `pytest`?

**Comment by harningle** [comment]
Thanks a lot!

**`lap_number` vs `number` renaming.** I like this very much. The old `number` is too vague: car No.? Lap No.? It makes sense to have a clearer lap number naming. But the output of this will feed to https://github.com/jolpica/jolpica-f1/issues/28 eventually, so please let me discuss all naming with him first.

**Test.** Yes we should test this. Can you make separate scripts to test them? We can hardcode (or get from Ergast) the correct data for some race(s), and test the function output using `pytest`? I tested it somehow (in a very bad way) in https://github.com/harningle/fia-doc/blob/main/notebook/cross_validate.ipynb. But we should put them in a proper testing rather than a random Jupyter notebook. If this is too heavy for one PR you can separate it into another one or leave it to me? I will open an issue on testing,

**Race Lap Parsing.** I don't like the current parsing and will probably change them. We have multiple PDFs that give us effectively the same race lap time data, e.g. race lap analysis seems to be a proper subset of race history chart, and we may want to combine them somehow. I'm going to open a new issue one this as well.

**Comment by tylerjthomas9** [comment]
> **Test.** Yes we should test this. Can you make separate scripts to test them? We can hardcode (or get from Ergast) the correct data for some race(s), and test the function output using `pytest`? I tested it somehow (in a very bad way) in https://github.com/harningle/fia-doc/blob/main/notebook/cross_validate.ipynb. But we should put them in a proper testing rather than a random Jupyter notebook. If this is too heavy for one PR you can separate it into another one or leave it to me? I will open an issue on testing,

I was going to add more comprehensive testing with pytest. I like these just for quick testing without running pytest, but I can move them all into a standard pytest setup.

**Comment by tylerjthomas9** [comment]
I added the basic tests to pytest, and a github action to run them. I am going to look at what you've done in `cross_validate` and try to expand the testing. It is probably worth doing this part in a separate PR. Let me know what you think.

File: parse_race_final_classification.py
Comment (harningle) [inline parse_race_final_classification.py:1]: maybe just delete `h`? I didn't realise that we are not using the height

**Comment by tylerjthomas9** [comment]
Ref https://github.com/harningle/fia-doc/issues/9

I tweaked the entry list parsing to avoid using camelot. I think other tools could be used, but I did a quick alternative using `pymupdf`.

**Comment by harningle** [comment]
Thanks a lot! I'm merging this. #9 may need some extra fine tuning as 2024 entry list comes in a slight different format. We can do that in another PR if you want.

**Pull request merged**